x/programs/cmd/simulator/vm/actions/program_execute.go

```diff
@@ -107,8 +106,12 @@ func (t *ProgramExecute) Execute(
 	importsBuilder.Register("state", func() host.Import {
 		return pstate.New(logging.NoLog{}, mu)
 	})
+	callContext := program.Context{
+		ProgramID:        programID,
+		Actor:            [32]byte(actor[1:]),
+		OriginatingActor: [32]byte(actor[1:])}
```

Comment:
You know what might be neat is passing the stack of actors rather than just parent actor and original actor 🤔

Reply:
Do you have use case ideas ?
It seems that if a contract is calling itself then this array would be redundant and linearly growing, making the execution and simulation more expensive. Also, it *may* encourage some bad patterns such as looking for every element in the array instead of the last one and the very first one.

Reply:
I have always found that the more data available during contract execution the better. I think contracts could enforce call depth or only certain calling "paths" (might be nice for security or something).